packages/babel-plugin-i18n-calypso/src/index.js

```diff
@@ -81,13 +94,56 @@ function getNodeAsString( node ) {
 	}
 }
 
+/**
+ * Returns true if the specified funciton name is valid translate function name
+ *
+ * @param {string} name Function name to test.
+ *
+ * @returns {boolean} Whether function name is valid translate function name.
+ */
+function isValidFunctionName( name ) {
+	return -1 !== Object.keys( DEFAULT_FUNCTIONS_ARGUMENTS_ORDER ).indexOf( name );
+}
+
+/**
+ * Returns true if the specified key of a function is valid for assignment in
+ * the translation object.
+ *
+ * @param {string} key Key to test.
+ *
+ * @returns {boolean} Whether key is valid for assignment.
+ */
+function isValidTranslationKey( key ) {
+	return Object.values( DEFAULT_FUNCTIONS_ARGUMENTS_ORDER ).some(
+		args => -1 !== args.indexOf( key )
+	);
+}
+
 module.exports = function() {
 	let strings = {},
 		nplurals = 2,
-		baseData;
+		baseData,
+		functions = { ...DEFAULT_FUNCTIONS_ARGUMENTS_ORDER };
 
 	return {
 		visitor: {
+			ImportDeclaration( path ) {
+				// Set functions.__ as reference to functions.translate if
+				// `translate` from  `i18n-calypso`` is aliased as `__`.
+				if (
+					'i18n-calypso' === path.node.source.value &&
+					path.node.specifiers.some(
+						specifier =>
+							specifier.imported &&
+							specifier.local &&
+							'translate' === specifier.imported.name &&
+							'__' === specifier.local.name
+					)
+				) {
+					functions.__ = functions.translate;
```

Comment:
Theoretically we could also remove the `'__' === specifier.local.name` from the `if` clause and use `functions[ specifier.local.name ] = functions.translate`, right? I wonder if maybe someone also renamed `_n` to `translate` which would actually work even better.

Reply:
Changed it to reference all `calypso-i18n` aliased imports in https://github.com/Automattic/wp-calypso/pull/39280/commits/4ff586547ceeac03df933d99274013cde3ce337e